File: PHP/uberVan.php
<?php
require_once('car.php'); # trae el archivo que hereda 
class UberVan extends Car{
    public $brand;
    public $model;
    public $typeCarAccepted;
    public $seatsMaterial;
    
    // protected $passenger; 

    public function __construct($license, $driver, $brand , $model, $typeCarAccepted, $seatsMaterial){
        parent::__construct($license,$driver);
        $this->typeCarAccepted = $typeCarAccepted;
        $this->seatsMaterial = $seatsMaterial;
        $this->brand = $brand;
        $this->model = $model;

    }

    public function setPassenger($passenger)
    {
        if ($passenger == 6){
            $this->passenger = $passenger;
        }
        else{
            echo "Necesitas asignar 4 pasajeros";
            echo "<br>"; // Salto de linea  
        }
    }

    public function printDataCar()
    {
        echo "Modelo: $this->model Brand: ".$this->brand;
        echo "<br>"; // Salto de linea 
    }    
}
?>

File: Java/UberVan.java
package Java;
import java.util.ArrayList;
import java.util.Map;



public class UberVan extends Car{
    Map<String, Map<String,Integer>> typeCarAccepted;
    ArrayList<String> seatsMaterial;
    private Integer passenger;

    public UberVan(String license, Account driver){
        super(license, driver);
    }

    public UberVan(String license, Account driver, Map<String, Map<String,Integer>> typeCarAccepted, 
    ArrayList<String> seatsMaterial){
        super(license, driver);
        this.typeCarAccepted = typeCarAccepted;
        this.seatsMaterial = seatsMaterial;
    }

    @Override // Sobrescribiendo este metodo de la clase Padre POLIMORFISMO 
    public void setPassenger(Integer passenger) {
        if(passenger == 6){ // condicion para que se cumple 
            this.passenger = passenger;
        }else{
            System.out.println("Necesitas asignar 6 pasajeros"); // mensaje de advertencia que no se cumple la condicion
        }
    }
}
